src/common/data-store/index.js

```diff
@@ -0,0 +1,95 @@
+import browser from "webextension-polyfill";
+
+import { encryptData, decryptData } from "../../common/lib/crypto";
+
+/**
+ * Encripted storage module. A passord is required to initialize this module.
+ * The storage is under the form of key-value pairs.
+ */
+let cachedStorage = null;
+
+function init(password) {
+  if (!password) {
+    throw new Error("A passord is required to initialize the storage module!");
+  }
+
+  const pwd = `${password}`;
+
+  function get(key) {
+    if (!key || typeof key !== "string") {
+      throw new Error("Invalid key: " + key);
+    }
+    return browser.storage.sync.get(key).then((result) => {
+      try {
+        if (!result || !result[key]) return;
+        const decriptedData = decryptData(result[key], pwd);
+        return decriptedData && decriptedData.data;
+      } catch (err) {
+        console.error(err);
+        throw new Error("Cannot decode data from storage. Wrong password!");
+      }
+    });
+  }
+
+  function set(key, data) {
+    if (!key || typeof key !== "string") {
+      throw new Error("Invalid key: " + key);
+    }
+
+    const encryptedData = encryptData({ data }, pwd);
```

Comment:
as discussed in the chat. 
I think encrypting every single storage data will prevent some features and will make data management harder. 
The only sensitive data we have are the ln account credentials, but allowances or settings would not need to be encrypted I think?